controllers/helpers.go

```diff
@@ -298,41 +302,136 @@ func GetSubnetUUID(ctx context.Context, client *prismclientv3.Client, peUUID str
 	return foundSubnetUUID, nil
 }
 
-// GetImage returns an image. If no UUID is provided, returns the unique image with the name.
+// GetImage returns an image UUID. If no UUID is provided, returns the unique image with the name.
 // Returns an error if no image has the UUID, if no image has the name, or more than one image has the name.
-func GetImage(ctx context.Context, client *prismclientv3.Client, id infrav1.NutanixResourceIdentifier) (*prismclientv3.ImageIntentResponse, error) {
-	switch {
-	case id.IsUUID():
-		resp, err := client.V3.GetImage(ctx, *id.UUID)
-		if err != nil {
-			if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
-				return nil, fmt.Errorf("failed to find image with UUID %s: %v", *id.UUID, err)
+func GetImage(
```

Comment:
Please, to keep the code easy to understand and test, let `GetImage` remain a way to get an image from a resource identifier, i.e., a UUID or name). 

The caller can decide whether to call `getImageWithFormat`, e.g., if `spec.ImageLookup != nil`, or `GetImage` otherwise.

We can export `getImageWithFormat` , and perhaps rename it to `GetMostRecentImageWithFormat`, to make its semantics clear.

Reply:
Interesting, so move the responsibility of deciding which field to use back out to the caller. I think that makes sense.

Reply:
Moved that logic out and left the function as is.